API, DOC: clarify "remove" command/function

stsievert opened the pull request:
**What does this PR implement?**
It does a couple things:

* Renames CLI command and HTMap function "remove" to "abort"
* Renames HTMap map attribute `is_removed` to `exists` (and adds documentation and changes tests to reflect boolean flip).
* Adds documentation note about `htmap abort`

**Reference issues/PRs**
This PR resolve #220.

stsievert commented:
I'm not sure what your deprecation policy is. I went ahead as if there was no deprecation policy for backwards-incompatible changes. Let me know if that should change.

JoshKarpel commented:
Since we're below v1, the deprecation policy is "mention it in the release notes".

I like `Map.is_removed` => `Map.exists`, but "Map.remove" is intended to mirror the underlying HTCondor operation, so I don't think we can change the name. I like the idea of explaining what it does in plain text in more places though, without assuming knowledge of the HTCondor operation of the same name.

stsievert commented:
> Map.remove is intended to mirror the underlying HTCondor operation

And that makes sense in the context of the HTCondor queue. I've added an FAQ towards that and reverted d22ba75a44d60f448a42d56f3732ceec68b288dc
